emcc.py

```diff
@@ -2934,6 +2934,8 @@ def do_binaryen(target, asm_target, options, memfile, wasm_binary_target,
                 optimizer):
   global final
   logger.debug('using binaryen')
+  if use_source_map(options) and not shared.Settings.SOURCE_MAP_BASE:
+    logger.warning("Wasm source map won't be usable in a browser without --source-map-base")
```

Comment:
Why move this error here?

Reply:
Earlier it was in the code that emits the source map. But that code might be called more than once (in particular with the wtmaps stuff). This puts the error in a single place where it's emitted exactly once.